File: compiler/testData/loadJava/kotlinSignature/propagation/typeParameter/TwoSuperclasses.java
package test;

import org.jetbrains.annotations.NotNull;
import jet.runtime.typeinfo.KotlinSignature;

public interface TwoSuperclasses {

    public interface Super1 {
        @KotlinSignature("fun <A: CharSequence> foo(a: A)")
        <A extends CharSequence> void foo(A a);
    }

    public interface Super2 {
        @KotlinSignature("fun <B: CharSequence> foo(a: B)")
        <B extends CharSequence> void foo(B a);
    }

    public interface Sub extends Super1, Super2 {
        <C extends CharSequence> void foo(C c);
    }
}

File: compiler/testData/loadJava/kotlinSignature/propagation/typeParameter/TwoSuperclasses.kt
package test

public trait TwoSuperclasses: Object {

    public trait Super1: Object {
        public fun <A: CharSequence> foo(p0: A)
    }

    public trait Super2: Object {
        public fun <B: CharSequence> foo(p0: B)
    }

    public trait Sub: Super1, Super2 {
        override fun <C: CharSequence> foo(p0: C)
    }
}
